classes/module/Module.php

```diff
@@ -1603,16 +1603,35 @@ public static function getModulesOnDisk($use_config = false, $logged_on_addons =
         return $module_list;
     }
 
+    /**
+     * @param \StdClass $modaddons Addons Module object, provided by XML stream
+     *
+     * @return string|null
+     */
     public static function copyModAddonsImg($modaddons)
     {
         if (!Validate::isLoadedObject($modaddons)) {
-            return;
+            return null;
         }
```

Comment:
Returning null is too silent ... 
Rather check if it is a module instance instead of isLoadedObject and throw an exception.

something like that :
```php
         if (!($modaddons instanceof Module)) {
            throw new InvalidModuleException('Invalud module ... etc etc');
        }
```

Reply:
I'm afraid of modifying the code of such an old class used so deeper 😅

This part of the code is very tangled and buried deep inside layers of layers of code. I'm afraid of breaking something.